### ISSUE TITLE
Graceful closedown process

### ISSUE THREAD

**Issue by JimKillock** [mastodon/mastodon#27334]
### Pitch

When an instance closes, typically some users have not migrated as they have missed notices. Instead of a simple one stage closedown process, a closedown could be done in stages, to allow late migrations and incentivise off boarding:

1. Stage one: notification (as it works at present)
2. Stage two: close to posting, boosting, replying to new messages.
3. Stage three (optional): closed to receiving any new content except for confirmation of migration signals
4. Stage four: final closedown

In this process, stage two would deactivate all features and behaviours except for receipt of messages, so that the platform does not appear broken to the outside world. However, users would be unable to use the site except passively, or to migrate their account to a new server.

In stage three, only migration out would be possible for the user. All other features including receipt of content would be switched off. This might need to work as content being discarded, in order that the server appears to be working to the outside world.

### Motivation

Giving users additional signals that the site is closing or has in effect closed would encourage users to migrate for longer.

**Comment by Openmedianetwork** [comment]
This is needed to avoid burn out on admins and mods who are likely close to burning out, otherwise the instances would not be closeting in the first place.

"You always have people who don't do things in good time, we don't get paid to do the work they missed (we don't get paid at all), the best we can do is be clear and help the community in general, there will always be people who fall between the cracks despite this, if #DIY culture was working that would be a "help" group to mediate this, we are not that group and will BURN out very fast if we try to be."

**Comment by nemobis** [comment]
Agreed something like this would be useful. Even step 1 is currently underdeveloped, see https://github.com/mastodon/mastodon/issues/19400 for a proposal.

**Comment by nemobis** [comment]
Also a data dump would simplify migrating entire instances: https://github.com/mastodon/mastodon/issues/19399 .
